### PR TITLE
Consider reversed operands order when comparing version info

### LINKED ISSUE #10264: Python version check is sensitive to comparison order
**Bug Report**

Whether mypy successfully detects a Python version check is dependent on the order of the comparison.

**To Reproduce**

```python
"""This will not pass on Python 3.8."""
import sys

if (3, 8) <= sys.version_info:
    pass
else:
    import invalid_library

```

```sh
% mypy type.py --python-version 3.8
type.py:6: error: Cannot find implementation or library stub for module named 'invalid_library'
type.py:6: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```

```python
"""This will pass on Python 3.8."""
import sys

if sys.version_info >= (3, 8):
    pass
else:
    import invalid_library
```

```sh
% mypy type.py --python-version 3.8
Success: no issues found in 1 source file
```

**Expected Behavior**

I expected mypy to correctly parse the Python version check regardless of the order of the comparison.

**Actual Behavior**

`(3, 8) <= sys.version_info` is not correctly detected by mypy, while the logically equivalent `sys.version_info >= (3, 8)` is.

**Your Environment**

- Mypy version used: 0.812
- Mypy command-line flags: `--python-version 3.8`
- Mypy configuration options from `mypy.ini` (and other config files): Empty
- Python version used: 3.8.6
- Operating system and version: macOS 10.15.7

### PR DESCRIPTION
**Pull request by kamilturek**
### Description

Closes #10264.

Consider reversed order of operands when trying to compare version info. When reversed order is used, operator is reversed as well for correct comparison.

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]

File: mypy/reachability.py [hunk 1]
```diff
@@ -127,10 +135,15 @@ def consider_sys_version_info(expr: Expression, pyversion: Tuple[int, ...]) -> i
...
         return TRUTH_VALUE_UNKNOWN
+
+    index = contains_sys_version_info(expr.operands[0])
     thing = contains_int_or_tuple_of_ints(expr.operands[1])
-    if thing is None:
+    if index is None or thing is None:
+        index = contains_sys_version_info(expr.operands[1])
+        thing = contains_int_or_tuple_of_ints(expr.operands[0])
+        op = reverse_op[op]
+    if index is None or thing is None:
```
Comment (JelleZijlstra) [inline mypy/reachability.py:145]: This check is redundant, since the conditions below won't trigger if either value is None.

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Please remove the redundant check.

File: mypy/reachability.py (hunk 1 above)
Comment (kamilturek) [reply mypy/reachability.py]: That's right. Good point 👍

**Review by JelleZijlstra — APPROVED** [review]

**Comment by kamilturek** [comment]
@JelleZijlstra One job on Travis CI has failed for some weird reason, doesn't look like thing related to this PR.

**Comment by JelleZijlstra** [comment]
Yeah that looks harmless. I'll merge this PR, thanks for your contributions!

**Pull request merged**